### PR TITLE
Update DeveloperGuide.md according to comments

### PR DESCRIPTION
**Pull request by isaacsaw25**
Updated according to comments on the [repo PR](https://github.com/nus-cs2113-AY2425S1/tp/pull/8)

### CODE REVIEW

File: docs/UML_Diagrams/AllocatorSequence.drawio.svg
Comment (ehz0ah) [inline docs/UML_Diagrams/AllocatorSequence.drawio.svg:1]: - Allocator class can be unbolded
- return arrow for some function calls are not aligned properly to the end of the activation bar

**Review by ThisisXXZ — APPROVED** [review]
Nice work adjusting the diagram according to the comments!

File: docs/UML_Diagrams/ParserSequence.drawio.svg
Comment (ehz0ah) [inline docs/UML_Diagrams/ParserSequence.drawio.svg:1]: What was changed here?

**Review by ehz0ah — CHANGES_REQUESTED** [review]
Just a few nits to fix (～￣▽￣)～

**Pull request merged**